contracts/Fund.sol

```diff
@@ -9,6 +9,11 @@ contract Fund {
     mapping(address => uint256) public amountFundedByAddress;
```

Comment:
I also suggest giving this different naming, let's call the contract e.g. `CollateralFunds` , so that the name of the contract is clear in itself, that it handles deals with saving collateral. Thus also the mapping on line 9, maybe better with `collateralFundedByAddress`